pkg/gatherers/clusterconfig/gather_openshift_config_apiserver.go

```diff
@@ -0,0 +1,58 @@
+package clusterconfig
+
+import (
+	"context"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/insights-operator/pkg/record"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GatherOpenshiftConfigAPIServer Collects APIServer.config.openshift.io resource
+//
+// ### API Reference
+// https://github.com/openshift/client-go/blob/master/config/clientset/versioned/typed/config/v1/apiserver.go
+//
+// ### Sample data
+// - docs/insights-archive-sample/config/apiserver.json
+//
+// ### Location in archive
+// - `config/apiserver.json`
+//
+// ### Config ID
+// `openshift_config_apiserver`
+//
+// ### Released version
+// - 4.15
+//
+// ### Backported versions
+// - TBD
+//
+// ### Changes
+// None
+func (g *Gatherer) GatherOpenshiftConfigAPIServer(ctx context.Context) ([]record.Record, []error) {
+	configClient, err := configv1client.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return configAPIServer{}.gather(ctx, configClient.APIServers())
+}
+
+type configAPIServer struct{}
```

Comment:
Is this type needed?

Reply:
for the logic? no. But I think it brings some readability and avoids charging clusterconfig package with more unexported functions.

Empty structs are practically free and this way of presenting the code encloses better the scope and gives you more flexibility on function naming.

But, I don't have a strong opinion on this, if you prefer I can change it to same name as the exported function.